xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java

```diff
@@ -228,9 +234,60 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     }
   }
 
+  @VisibleForTesting
+  static class EdsLoadBalancingHelper extends ForwardingLoadBalancerHelper {
+    private final Helper delegate;
+    private final AtomicReference<UpstreamTlsContext> upstreamTlsContext;
+
+    EdsLoadBalancingHelper(Helper helper, AtomicReference<UpstreamTlsContext> upstreamTlsContext) {
+      this.delegate = helper;
+      this.upstreamTlsContext = upstreamTlsContext;
+    }
+
+    @Override
+    public Subchannel createSubchannel(CreateSubchannelArgs createSubchannelArgs) {
+      if (upstreamTlsContext.get() != null) {
+        createSubchannelArgs =
+            createSubchannelArgs
+                .toBuilder()
+                .setAttributes(createSubchannelArgs.getAttributes())
```

Comment:
This is not needed now.

Reply:
I need to copy attributes from the original `createSubchannelArgs` to the new one I am building - otherwise I could be breaking somebody else's code. Note I am not adding anything.

Reply:
It's already copied by default I believe.

Reply:
you are right - I missed that!